### ISSUE TITLE
LDRedisOptions typing bug

### ISSUE THREAD

**Issue by mikewesthad** [launchdarkly/js-core#479]
**Is this a support request?**
No

**Describe the bug**

The [LDRedisOptions](https://github.com/launchdarkly/js-core/blob/main/packages/store/node-server-sdk-redis/src/LDRedisOptions.ts#L16) exported don't match [the ioredis constructor options](https://github.com/redis/ioredis/blob/v5.3.1/lib/Redis.ts#L120). Specifically, a redis:// string is an acceptable option to construct a ioredis instance. This is how we connect to redis when using LD so it would be ideal if this were supported and we didn't need to do a ts ignore error:

```ts
const featureStore = RedisFeatureStore({
  // @ts-expect-error LD's wrapper around ioredis doesn't have the correct types
  // - a string URL with username, password and port is acceptable.
  redisOpts: redisUrl,
  prefix: `launchdarkly:replit:${process.env.NODE_ENV}`,
  cacheTTL: 30,
});
```

**To reproduce**
See above

**Expected behavior**
Expect to be able to connect using the same options that `ioredis` supports

**Logs**
N/A

**SDK version**
```
"@launchdarkly/node-server-sdk": "~8.2.6",
"@launchdarkly/node-server-sdk-redis": "^4.1.14",
```

**Language version, developer tools**
Node 20

**OS/platform**
Ubuntu

**Additional context**
N/A

**Comment by kinyoklion** [comment]
Hello @mikewesthad,

We support the `RedisOptions` directly from `ioredis`, but we do not support the various short-hand constructor options.

Alternatively you can omit the `redisOpts` and directly construct a client and pass it to the `client` parameter.

Thank you,
Ryan

**Comment by kinyoklion** [comment]
@mikewesthad Which redis library are you using? 

Under the hood it currently required ioredis, so it may get encapsulated, but currently it will be there either way. 

We had moved to ioredis because it was more capable and widely used than the library we had used previously. But if there are sufficient users and capabilities, then we could consider additional integrations in the future.

Thank you,
Ryan
